### DISCUSSION TITLE
pickup a cargo on visit in a mission

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#9475] [General]
Greetings,
I want to create a mission where the play flies to system A and if he visits planet B in that system, he picks up and item as cargo.
seems like the cargo entry is not supported under on visit directive.
so I was wondering what is the proper way to do so?

**Comment by GPaso** [comment]
Do you want to give the player cargo as in commodities or an outfit (as cargo) ?

**Reply by daggs1** [reply]
custom cargo

**Reply by GPaso** [reply]
Indeed it doesn't seems possible unless you create another mission like @quyykk suggested.
but i think it should be implemented.

**Reply by MasterOfGrey** [reply]
It's a little difficult, because the mission would have to be able to do all the things that missions currently do when offered, on a stopover.
Otherwise, you could be given cargo above your capacity somewhere that you can't fix that, or you'd need to have space reserved from the start of the mission.
So it would be a lot of steps to implement actually getting cargo from a stopover mid-way into a mission.

**Reply by GPaso** [reply]
Got it, so it would need to check for available cargo and even display a message if the cargo has been loaded or not. It is a bit more of what i had thougth. Maybe just easyer to make multiple missions.

**Reply by daggs1** [reply]
assuming I split it to two missions, how can I auto accept the second mission when the first one is completed?

**Comment by quyykk** [comment]
I believe the usual way to do this is to make two missions, one to fly to system A and another one that offers when you land on planet B.

**Reply by daggs1** [reply]
infact, there is an already such mission see: https://github.com/endless-sky/endless-sky/blob/master/data/human/deep%20jobs.txt#L288C23-L288C32
I'll try that format and see if it works
